symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/BotInfoService.java

```diff
@@ -8,15 +8,15 @@
 import lombok.extern.slf4j.Slf4j;
 
 /**
- * Base class for implementing the bot ingo services.
+ * Base class for implementing the bot info services.
  * BotInfoService can help the bot retrieving the information about itself by using session API.
  */
 @Slf4j
 public class BotInfoService {
 
-    private SessionApi sessionApi;
     private final AuthSession authSession;
-    private static UserV2 userInfo;
+    private SessionApi sessionApi;
+    private UserV2 userInfo;
```

Comment:
Here the userInfo is retrieved once and then never refreshed, is it ok for the usage we have today?
Is there a risk the user misuses this?

Reply:
IMHO, the information of a service account should not be changed during the bot is running. That's why I think it need to be retrieved only once.

Reply:
Actually do we need to call the infoservice? don't we have the username available in the bot's configuration and in the event we receive? (to avoid calling the API)

Reply:
I also thought about it but didn't find another way

Reply:
@symphony-hong : in class `AbstractDatafeedService`, could you try to change the method `isSelfGeneratedEvent` by : 

```java
    private boolean isSelfGeneratedEvent(V4Event event) throws ApiException {
        return event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername());
    }
```

In this case, call to the infoservice is not necessary.